libs/blocks/region-nav/region-nav.js

```diff
@@ -0,0 +1,42 @@
+import { getConfig } from '../../utils/utils.js';
+
+/* c8 ignore next 11 */
+function handleEvent(prefix, link, config) {
+  document.cookie = `international=${prefix};path=/`;
+  sessionStorage.setItem('international', prefix);
+  fetch(link.href, { method: 'HEAD' }).then((resp) => {
+    if (!resp.ok) throw new Error('request failed');
+    window.location.assign(link.href);
+  }).catch(() => {
+    const prefixUrl = prefix ? `/${prefix}` : '';
+    window.location.assign(`${prefixUrl}${config.contentRoot || ''}/`);
+  });
+}
+
+function decorateLink(link, config, path) {
+  let pathname = link.getAttribute('href');
+  if (pathname.startsWith('http')) {
+    try { pathname = new URL(pathname).pathname; } catch (e) { /* href does not contain domain */ }
+  }
+  const linkParts = pathname.split('/');
+  const prefix = linkParts[1] || 'us';
+  let { href } = link;
+  if (href.endsWith('/')) href = href.slice(0, -1);
+  link.href = `${href}${config.contentRoot || ''}${path}`;
+  link.addEventListener('click', (e) => {
+    /* c8 ignore next 2 */
+    e.preventDefault();
+    handleEvent(prefix, link, config);
+  });
+}
+
+export default function init(block) {
+  const config = getConfig();
+  const divs = block.querySelectorAll(':scope > div');
+  if (divs.length < 2) return;
+  const links = divs[1].querySelectorAll('a');
```

Comment:
shouldn't it be: divs[1].querySelectorAll(':scope a')?

Reply:
In this context the query result is the same. I'm calling querySelectorAll on the div, which only returns elements that are descendants of the div.
![image](https://user-images.githubusercontent.com/115231412/219046585-99955154-81f3-4d1e-a27f-a2ccfb084d7b.png)

Reply:
According to the MDN docs [0] `querySelectorAll` works differently, it doesn't return descendants of the element it is applies on, which can sometimes lead to unexpected results:
```
The querySelectorAll() method applies its selectors to the whole document: they are not scoped to the element on which the method is called. To scope the selectors, include the [:scope](https://developer.mozilla.org/en-US/docs/Web/CSS/:scope) pseudo-class at the start of the selector string.
```
[0] https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelectorAll

Reply:
not quite, it does only return elements that are descendants of the div, which is what I want. Look at the first three sentences on the page you linked.  
The selectors do apply to the whole document though, which is what you're saying, but that doesn't matter in my case. The cases are quite specific to have unexpected results, and to be honest, even those unexpected links I'd want to have since they will be direct descendants of the div.

Reply:
you're right, I was mislead by the comment on `:scope`. I assume that using `:scope` is a bit more performant as it only applies its selectors on the element on which the method is called, not on the whole document. It's probably not a big deal in your case, so it's fine to keep the code it as is.